sycl/test-e2e/Graph/AsyncAlloc/lit.local.cfg

```diff
@@ -0,0 +1,5 @@
+# Async alloc tests require virtual memory for device allocations
+# Async alloc extension is required to support creating memory pools
+config.required_features += ['aspect-ext_oneapi_virtual_mem', 'aspect-ext_oneapi_limited_graph', 'aspect-ext_oneapi_async_memory_alloc']
```

Comment:
What kind of error gets thrown if a device doesn't have support for virtual mem or async mem alloc but a user tries to use allocation nodes in a graph?

Reply:
Virtual mem throws `feature_not_supported` when you try to add an allocation node if not supported. 

Async mem is a bit trickier since it doesn't check specifically in the SYCL runtime but rather relies on whether the UR call (which we bypass) is implemented or not (outside of generally yet to be implemented features like host/shared allocations). So currently it will throw if the user tries to make a memory pool for an unsupported backend, but graph allocations not using memory pools will work even if the backend doesn't support async alloc in UR (if it supports virtual memory).

Reply:
cool, I think that's fine. My main concern was users having a bad experience by not getting unsupported errors early, but that doesn't sounds like it's an issue.